### ISSUE TITLE
Ability to split special tokens with fast tokenizer

### ISSUE THREAD

**Issue by scruel** [huggingface/transformers#31337]
### Feature request

The following workaround will work for same goal:
```
tokenizer._tokenizer.encode_special_tokens=False
```

but the following won't work:
```
tokenizer = AutoTokenizer.from_pretrained(
"meta-llama/Llama-2-7b-hf", local_files_only=True, split_special_tokens=True)
```

### Motivation

`split_special_tokens` param now only supports the slow tokenizer

### Your contribution

/

**Comment by scruel** [comment]
The following workaround will work for same goal:
```
tokenizer._tokenizer.encode_special_tokens=False
```

**Comment by scruel** [comment]
Already fixed with [deba765](https://github.com/huggingface/transformers/commit/deba7655e6e54fb885e79204dec9f767393dd2df)
